docs/development/workflow/git_edit_workflow_examples.rst

```diff
@@ -52,98 +43,101 @@ Grab the latest updates to astropy
 A few steps in this tutorial take only a single command. They are broken out
 separately to outline the process in words as well as code.
 
-Inform your local copy of Astropy about the latest changes in the development
+Inform your local copy of ``astropy`` about the latest changes in the development
 version with::
 
     git fetch astropy --tags
 
 Set up an isolated workspace
 ============================
+
 + Make a new `git`_ branch for fixing this issue, check it out, and let my
   GitHub account know about this branch::
 
-    git branch fix-1761 astropy/master   # branch based on latest from GitHub
-    git checkout fix-1761                   # switch to this branch
-    git push --set-upstream origin fix-1761 # tell my github acct about it
+    git checkout astropy/master -b fix-1761 # branch based on latest from GitHub
+                                            # and switch to this branch
+    git push your-user-name fix-1761        # tell my GitHub acct about it
```

Comment:
I'm confused why this is here (and in the original). If you do it later, when the work is ready, then github will give you a nice web link to copy & paste to do the PR.

Also, if you keep it here, I would keep the `--set-upstream`...

Reply:
for what it's worth, I never manually say `--set-upstream`. Otherwise I agree that it might be better to push only once some work is done to get the github pop-up for opening the PR (maybe it also works when pushing for an existing branch?, I've never tried).

Reply:
I never use `--set-upstream`. In fact, I think a plain old `git push` is dangerous (personal opinion) and prefer to be explicit when I do the push. If there is a strong preference to add it back, I will.

Sure, I can also delay `git push` till the end. I wasn't questioning why this tutorial was pushing so often.

Reply:
oh, `git push`, just as is, is only asking for the troubles, I don't think should advocate for it anywhere.

Reply:
I do use `-u` (I find I'm more likely to push to the wrong branch if I have to enter a name, even with auto-completion...), but for the dev workflow example, I agree it is better not to - keep things as simple as possible. And really it is only useful if one keeps on pushing. So, if you are up for it, I would remove the git push until after the user has done the testing. After all, the whole point of git is to enable asynchronous development - there is no reason to keep github in the loop.

Reply:
@mhvk - I use gitprompt that decorates my bash prompt (there are way more fancy solutions, too, I just never got to the point of switching), so I always see which branch I'm at. It gives a very convenient double check for names, as well as branch status compared to upstream/origin. And I just got into the habit of double checking with a git gui whenever I need to force push to a branch (in those cases I usually just see the hash as normally I don't bother to create a branch when working with other people's forks). 
Overall I find this a highly efficient, but also a quite safe setup.

Reply:
I am not sure if we want to get in the weeds on setting up custom git prompt etc. But I did remove all the extraneous `git push` commands.

Reply:
Oh, I'm not saying we should do anything about git prompt, it's on the same level of arbitrary personal preferences as the choice of editor. It was only a pointer to Marten's point that helps to avoid pushing to the wrong branch, in case he doesn't use it or something similar.